R/get-path-from-object.R

```diff
@@ -158,45 +158,234 @@ get_model_id.bbi_model <- function(.mod) {
 
 #' Get path to data file
 #'
-#' Helper to extract the path to the input data from a
-#' `bbi_{.model_type}_summary` or `bbi_{.model_type}_model` object. Note that
-#' **this function only works on models that have been successfully run**
-#' because the path to the data is constructed during model submission.
-#' @param .mod Model to use, either a `bbi_{.model_type}_model` or
-#' `bbi_{.model_type}_summary` object.
+#' Helper to extract the path to the input data from a bbi object.
+#'
+#' @param .bbi_object The object to query. Could be
+#' a `bbi_{.model_type}_model` object,
+#' `bbi_{.model_type}_summary` object,
+#' or a tibble of class `bbi_log_df`.
+#' @param .check_exists If `TRUE`, the default, will throw an error if the file
+#'  does not exist
 #' @param ... Arguments passed through to methods. (Currently none, but may have
 #'   some modifier arguments other model types in the future.)
-#' @importFrom fs path_rel
+#'
+#' @template nonmem-mod-ext
+#' @seealso [get_model_id()] [get_path_from_object()] [build_path_from_model()]
+#'
 #' @return Absolute path to input data file
 #' @export
-get_data_path <- function(.mod, ...) {
+get_data_path <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    ...
+){
   UseMethod("get_data_path")
 }
 
-#' @describeIn get_data_path Takes `bbi_{.model_type}_model` or `bbi_{.model_type}_summary` object
+#' @rdname get_data_path
 #' @export
-get_data_path.bbi_model <- function(.mod, ...) {
+get_data_path.bbi_model <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    ...
+){
+  data_path <- get_data_path_from_json(.bbi_object)
+  if(.check_exists){
```

Comment:
Minor, but best for this to use `isTRUE` because this value is coming from the caller as is. (Same comment may apply in some other spots here.)